src/main/kotlin/platform/mixin/handlers/mixinextras/MixinExtrasInjectorAnnotationHandler.kt

```diff
@@ -0,0 +1,338 @@
+                val ownerType = Type.getObjectType(insn.owner).toPsiType(elementFactory)
+                val sourceClassAndField = (
+                    MemberReference(insn.name, insn.desc, insn.owner.replace('/', '.'))
+                        .resolveAsm(annotation.project) as? FieldTargetMember
+                    )?.classAndField
+                val valueType =
+                    sourceClassAndField?.field?.getGenericType(sourceClassAndField.clazz, annotation.project)
+                        ?: Type.getType(insn.desc).toPsiType(elementFactory)
+                val owner = Parameter("instance", ownerType)
+                val value = Parameter("value", valueType)
+                when (insn.opcode) {
+                    Opcodes.GETFIELD -> listOf(owner)
+                    Opcodes.PUTFIELD -> listOf(owner, value)
+                    Opcodes.GETSTATIC -> emptyList()
+                    Opcodes.PUTSTATIC -> listOf(value)
+                    else -> null
+                }
+            }
+
+            is TypeInsnNode -> {
+                when (insn.opcode) {
+                    Opcodes.INSTANCEOF ->
+                        listOf(Parameter("object", Type.getType(Any::class.java).toPsiType(elementFactory)))
+
+                    else -> null
+                }
+            }
+
+            else -> null
+        } ?: getInsnArgTypes(insn, targetClass)?.map { Parameter(null, it.toPsiType(elementFactory)) }
+    }
+}
+
+private val CONSTANTS_ALL = intArrayOf(
+    Opcodes.ACONST_NULL,
+    Opcodes.ICONST_M1,
+    Opcodes.ICONST_0, Opcodes.ICONST_1, Opcodes.ICONST_2, Opcodes.ICONST_3, Opcodes.ICONST_4, Opcodes.ICONST_5,
+    Opcodes.LCONST_0, Opcodes.LCONST_1,
+    Opcodes.FCONST_0, Opcodes.FCONST_1, Opcodes.FCONST_2,
+    Opcodes.DCONST_0, Opcodes.DCONST_1,
+    Opcodes.BIPUSH, // 15
+    Opcodes.SIPUSH, // 16
+    Opcodes.LDC, // 17
+    Opcodes.CHECKCAST, // 18
+    Opcodes.INSTANCEOF // 19
+)
+
+private val CONSTANTS_TYPES = arrayOf(
+    "V", // null is returned as Type.VOID_TYPE
+    "I",
+    "I", "I", "I", "I", "I", "I",
+    "J", "J",
+    "F", "F", "F",
+    "D", "D",
+    "I", // "B",
+    "I"
+)
+
+private fun isConstant(insn: AbstractInsnNode?): Boolean {
+    return insn != null && insn.opcode in CONSTANTS_ALL
+}
+
+private fun getConstantType(insn: AbstractInsnNode?): Type? {
+    return when (insn) {
+        null -> null
+        is LdcInsnNode -> {
+            val cst = insn.cst
+            when (cst) {
+                is Int -> return Type.INT_TYPE
+                is Float -> return Type.FLOAT_TYPE
+                is Long -> return Type.LONG_TYPE
+                is Double -> return Type.DOUBLE_TYPE
+                is String -> return Type.getType(String::class.java)
+                is Type -> return Type.getType(Class::class.java)
+                else -> null
+            }
+        }
+
+        is TypeInsnNode -> {
+            return if (insn.getOpcode() < Opcodes.CHECKCAST) {
```

Comment:
Super pedantic, but here I'd be worried that in the future they will add new instructions >`CHECKCAST` that are a `TypeInsnNode`, that Mixin doesn't treat as constants.

Reply:
This whole method is copied from Mixin including that check

Reply:
Ah, whatever then. If new instructions do emerge and Mixin does get broken, we will at least be consistent with Mixin's broken behavior.